src/Microsoft.Restier.Core/ApiBase.cs

```diff
@@ -173,15 +190,11 @@ protected virtual ApiConfiguration CreateApiConfiguration(ApiBuilder builder)
         /// </param>
         protected virtual void Dispose(bool disposing)
         {
+            this.IsDisposed = true;
```

Comment:
Disposing the scope will get every service in the scope to be disposed, including this (IApi). So actually `ApiBase.Dispose()` will get a recursive call from ServiceScope, put the assignment before disposing the scope prevents `ApiConfiguratorAttribute.ApplyDisposal` called twice.